lib/schema.js

```diff
@@ -1504,6 +1504,49 @@ Schema.prototype.remove = function(path) {
   }
 };
 
+/**
+ * Loads an ES6 class into a schema. Maps setters + getters, static methods, and instance methods to schema virtuals, statics, and methods.
+ *
+ * @param {Function} model
+ */
+Schema.prototype.loadClass = function(model, virtualsOnly) {
+  if (model === Object.prototype || model === Function.prototype) {
+    return this;
+  }
+
+  // Add static methods
+  if (!virtualsOnly) {
```

Comment:
I suppose it's probably unnecessary. Its only use is [mainly internal by `Model.compile`](https://github.com/Automattic/mongoose/pull/4668/files/27d2342456bf366706842e2510db9f1148f52322#diff-a140b60b6a99a8c69f4b5d0d1f1f2bfaR3214). That's also why I didn't add JSDoc for the parameter.

It seems like Schemas *need* the virtuals defined on them for them to function properly, e.g. in `toObject` and `toJSON`. It just felt wasteful copying all of the statics and instance methods from the class onto the schema when `applyMethods(model, schema)` and `applyStatics(model, schema)` are being called immediately afterward.